bundle/jsky.coords/src/main/java/jsky/coords/HMS.java

```diff
@@ -166,42 +174,24 @@ public void setVal(double val) {
      * The seconds are formatted with 3 digits of precision.
      */
     public String toString() {
-        String secs = nf.format(sec);
-
-        // sign
-        String signStr;
-        if (sign == -1)
-            signStr = "-";
-        else
-            signStr = "";
-
-        return signStr
-                + nf.format(hours)
-                + ":"
-                + nf.format(min)
-                + ":"
-                + secs;
+        return toString(true);
     }
 
     /**
      * Return the value as a String in the form hh:mm:ss.sss,
      * or if showSeconds is false, hh:mm.
      */
     public String toString(boolean showSeconds) {
-        if (showSeconds)
-            return toString();
-
-        // sign
-        String signStr;
-        if (sign == -1)
-            signStr = "-";
-        else
-            signStr = "";
-
-        return signStr
-                + nf.format(hours)
+        return toString(showSeconds, true);
+    }
+
+    public String toString(boolean showSeconds, boolean showFractionalSeconds) {
+        final NumberFormat nf = showFractionalSeconds ? nf_frac : nf_noFrac;
+        return (sign == -1 ? "-1" : "")
+                + nf_noFrac.format(hours)
                 + ":"
-                + nf.format(min);
+                + nf_noFrac.format(min)
+                + (showSeconds ? (":" + nf.format(sec)) : "");
```

Comment:
It seems like if we're not showing seconds then minutes would either be optionally formatted as a decimal value or else rounded.  This whole class seems kind of sketchy.